docs/zephyr_02_functional_requirements.sdoc

```diff
@@ -1317,6 +1317,248 @@ As a Zephyr OS user I want to... ???,
 Architectural & interface REQ
 <<<
 
```

Comment:
General NitPick;  
Too much Console Font gives me migraines.

The titles can be simplified. 
- Remove "Counting Semaphore" since it is already in the COMPONENT above.
- Remove verbiage that is in the STATEMENT below, except the minimum needed to make the title unique.

Some examples: 

Counting Semaphore Exposes Current Count  
Get Current Count

Counting Semaphores Exposes Ability to Reset Count to Zero
Reset Count

Counting Semaphore Give unblocks pending Take by another thread
Give with pending Take

Counting Semaphore Take when Count is > 0
Take when Count is > 0

Reply:
Understood. Here is why I wrote them the way I did:

1. In general, each requirement document needs to be extensible (whether by internal or external future additions). Making the Title text unique and meaningful within an open set is just another case of namespace management.
2. As laid out now, the low-level requirements document is extremely broad. I expect someone (in addition to me) will import the entire document into another commercial tool and work with it there. I have worked with a few different ones and searches return the item titles without the section context. The "repetitive" information is for usability in that context. (NOTE: Hovering over the title in those GUIs may bring the description up in a tool tip, but sometimes the requirement statement is larger than the tool tip can display. Some CA Title 24 Building Energy Code regulation statements are well over 100 words long even after breaking them into singular statements.)
3. We must expect these requirements will be inherited as part of a down-stream product. What may be ubiquitous here may not be in the shipping product. At this point everything we are considering is still within the Kernel. To be reusable the requirements must be scoped accordingly. This will become more apparent when we start to include functionality outside the kernel (e.g., HALs, drivers, logging?, build system).

I think breaking the "low-level" requirements document into a separate document for each component (kernel object) will help greatly. Searches can be limited to the document and missing links will be far easier to identify. At that point the titles can be simplified.

Reply:
BTW, the reference I'm using for the architecture process is [here](http://swebokwiki.org/Chapter_2:_Software_Design#Software_Structure_and_Architecture).